web3.js/src/address-lookup-table-program.ts

```diff
@@ -0,0 +1,270 @@
+import * as BufferLayout from '@solana/buffer-layout';
+
+import * as Layout from './layout';
+import { PublicKey } from './publickey';
+import { SystemProgram } from './system-program';
+import { TransactionInstruction } from './transaction';
+import { encodeData, IInstructionInputData } from './instruction';
+
+export type CreateLookupTableParams = {
+    lookupTable: PublicKey;
+    authority: PublicKey;
+    payer: PublicKey;
+    slot: number;
+    bumpSeed: number;
+};
+
+export type FreezeLookupTableParams = {
+    lookupTable: PublicKey;
+    authority: PublicKey;
+};
+
+export type ExtendLookupTableParams = {
+    lookupTable: PublicKey;
+    authority: PublicKey;
+    payer: PublicKey;
+    addresses: Array<PublicKey>;
+};
+
+export type DeactivateLookupTableParams = {
+    lookupTable: PublicKey;
+    authority: PublicKey;
+};
+
+export type CloseLookupTableParams = {
+    lookupTable: PublicKey;
+    authority: PublicKey;
+    recipient: PublicKey;
+};
+
+export type LookupTableInstructionType =
+    | 'Create'
+    | 'Extend'
+    | 'Close'
+    | 'Freeze'
+    | 'Deactivate';
+
+export type LookupTableInstructionInputData = {
+    Create: IInstructionInputData &
+        Readonly<{
+            slot: number;
+            bumpSeed: number;
+        }>;
+    Freeze: IInstructionInputData;
+    Extend: IInstructionInputData &
+        Readonly<{
+            numberOfAddresses: number;
+            addresses: Array<Uint8Array>;
+        }>;
+    Deactivate: IInstructionInputData;
+    Close: IInstructionInputData;
+};
+
+export const LOOKUP_TABLE_INSTRUCTION_LAYOUTS = Object.freeze({
+    Create: {
+        index: 0,
+        layout: BufferLayout.struct<LookupTableInstructionInputData['Create']>([
+            BufferLayout.u32('instruction'),
+            BufferLayout.nu64('slot'),
+            BufferLayout.u8('bumpSeed'),
+        ]),
+    },
+    Freeze: {
+        index: 1,
+        layout: BufferLayout.struct<LookupTableInstructionInputData['Freeze']>([
+            BufferLayout.u32('instruction'),
+        ]),
+    },
+    Extend: (numberOfAddresses: number) => {
+        return {
+            index: 2,
+            layout: BufferLayout.struct<
+                LookupTableInstructionInputData['Extend']
+            >([
+                BufferLayout.u32('instruction'),
+                BufferLayout.nu64('numberOfAddresses'),
```

Comment:
Please the `u64` util here as well. Wish I would have made this a `u8` :/

Reply:
Fixed.